types/geojson/index.d.ts

```diff
@@ -160,7 +168,17 @@ export interface Feature<G extends GeometryObject, P = GeoJsonProperties> extend
  * A collection of feature objects.
  *  https://tools.ietf.org/html/rfc7946#section-3.3
  */
-export interface FeatureCollection<G extends GeometryObject, P = GeoJsonProperties> extends GeoJsonObject {
+export interface FeatureCollection<G extends GeometryObject = GeometryObject, P = GeoJsonProperties> extends BaseGeoJsonObject {
     type: "FeatureCollection";
     features: Array<Feature<G, P>>;
 }
+
+/**
+ * GeoJSON object to be used when accepting aribtrary GeoJSON
+ * https://tools.ietf.org/html/rfc7946#section-3
+ * The GeoJSON specification also allows foreign members
```

Comment:
Where do you see the advantage of this more specified type definition? Keep in mind that it is allowed to add foreign members...

Reply:
My goal with a more specific type is to reduce work when matching GeoJSON objects (please see the new test, shown in the original PR text). AFAICT, the existing type definitions force developers to write a number type narrowing functions when matching existing objects in conditions and `switch` statements.

It was not my intent to change any behavior around [Section 6.1: Foreign Members](https://tools.ietf.org/html/rfc7946#section-6.1). I note that adding foreign members on the `master` branch version of the definition results in errors. 

```ts
const testForeignMembers: Feature<Point> = {
  type: "Feature",
  id: "f2",
  geometry: {
    type: "Point",
    coordinates: [0, 0]
  },
  properties: {
    foo: "bar",
    hello: "world"
  },
  centerline: {
    type: "LineString",
    coordinates: [
      [-170, 10],
      [170, 11]
    ]
  }
}
```

```
Type '{ type: "Feature"; id: string; geometry: { type: "Point"; coordinates: number[]; }; properties: {...' is not assignable to type 'Feature<Point, { [name: string]: any; }>'. Object literal may only specify known properties, and 'centerline' does not exist in type 'Feature<Point, { [name: string]: any; }>'.
```

I note that the comments `BaseGeoJsonObject` (originally `GeoJsonObject`) states: `Developers should use "&" type in TypeScript or extend the interface to add these foreign members.` 

In both the original and updated, the following is valid:

```ts
interface foreignMembers {
  centerline: object,
}

const testForeignMembers: foreignMembers & GeoJsonObject = {
  type: "Feature",
  id: "f2",
  geometry: {
    type: "Point",
    coordinates: [0, 0]
  },
  properties: {
    foo: "bar",
    hello: "world"
  },
  centerline: {
    type: "LineString",
    coordinates: [
      [-170, 10],
      [170, 11]
    ]
  }
}
```

Note that this code is also valid for `const testForeignMembers: foreignMembers & Feature` (instead of `foreignMembers & GeoJsonObject`).

It may be in our interest to add a test for Foreign Members. I would be happy to update the PR with the code above if that is of interest.